app/frontends/stac.ts

```diff
@@ -60,13 +79,33 @@ export async function getStacCatalog(req, res): Promise<void> {
   const keys = keysToLowerCase(req.query);
   const linkType = keys.linktype?.toLowerCase();
   try {
-    await handleStacRequest(req, res, (data) => stacCatalogCreate(data, linkType));
+    const pagingParams = getPagingParams(req, env.defaultResultPageSize);
+    await handleStacRequest(
+      req, res,
+      (job: Job, pagingLinks: JobLink[]) => stacCatalogCreate(
+        job.jobID, job.request, job.links, pagingLinks, linkType,
+      ), pagingParams,
+    );
   } catch (e) {
     req.context.logger.error(e);
-    res.status(500);
-    res.json({
-      code: 'harmony:ServerError',
-      description: `Error: Internal server error trying to retrieve STAC catalog for job ${jobId}` });
+    if (e instanceof RangeError) {
```

Comment:
We generally let our middleware handle creating the appropriate JSON and just throw exceptions that are defined in `app/util/errors.ts`.

E.g.
```
if (e instanceof RangeError) {
  throw new RequestValidationError(e.message)
} ...
```

Reply:
Ah OK. Which middleware handler in particular? I was just trying to follow what the other route handler was doing (getStacItem): https://github.com/nasa/harmony/blob/main/app/routers/router.ts#L206 There were a lot of tests written that expect these custom responses.

Reply:
(`getStacCatalog` is a middleware function too).

https://github.com/nasa/harmony/blob/2f38b7f5beb5254f43af8a7265627cd0d20c9cb9/app/routers/router.ts#L205

I might be misunderstanding what you're saying.

Reply:
The middleware that formats the error messages is `app/middleware/error-handler.ts`. All other code should just throw the correct exception type with an appropriate message and the error handler will handle the presentation to the end user.

It does create messages with a 'code' and a 'description' when returning json errors. Given that the existing STAC code wasn't throwing exceptions to be handled by the error handler middleware it's probably out of scope for this ticket to refactor.

Reply:
I'll refactor it.